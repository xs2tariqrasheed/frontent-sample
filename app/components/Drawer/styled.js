import styled, { css } from 'styled-components';
import mq from '../../styles/mq';
import theme from '../../styles/theme';

export const Panel = styled.div`
  background: #322c8f;
  width: ${props => (props.isLarge ? '100%' : '100%')};
  max-width: 100vw;
  position: fixed;
  overflow: auto;
  top: 0;
  bottom: 0;
  color: ${theme.color.text};
  z-index: 50;
  outline: none;

  ${props =>
    props.isRight &&
    css`
      left: ${props.isOpen ? '0' : '-56rem'};
    `}
  ${props =>
    !props.isRight &&
    css`
      right: ${props.isOpen ? '0' : '-200rem'};
    `}

  transition: all 0.3s;

  ${mq.xs} {
    ${props =>
      props.isLarge && // eslint-disable-line
      css`
        width: 100%;
      `};
  }
`;
