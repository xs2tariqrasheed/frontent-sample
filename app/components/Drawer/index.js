import React from 'react';
import ModalPortal from 'components/ModalPortal';
import PropTypes from 'prop-types';
import { Panel } from './styled';

const Drawer = props => (
  <ModalPortal>
    <Panel
      isOpen={props.isOpen}
      isRight={props.isRight}
      isLarge={props.isLarge}
    >
      {props.children}
    </Panel>
  </ModalPortal>
);

Drawer.propTypes = {
  children: PropTypes.any,
  isOpen: PropTypes.bool,
  isRight: PropTypes.bool,
  isLarge: PropTypes.bool,
  // onClose: PropTypes.any,
};

export default Drawer;
export const LocklessDrawer = Drawer;
