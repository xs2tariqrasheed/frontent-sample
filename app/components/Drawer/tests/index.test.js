import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import Drawer from '../index';

describe('<Drawer />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Drawer isOpen isRight isLarge />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
