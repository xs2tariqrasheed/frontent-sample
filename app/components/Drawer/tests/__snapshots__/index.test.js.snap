// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should render and match the snapshot 1`] = `null`;
